subprojects/core/src/main/java/org/gradle/internal/operations/notify/BuildOperationNotificationBridge.java

```diff
@@ -97,7 +98,7 @@ public void stop() {
         @Override
         public void beforeSettings(Settings settings) {
             if (settings.getGradle().getParent() == null) {
-                settings.getGradle().projectsLoaded((InternalAction<Gradle>) project -> {
+                ((GradleInternal)settings.getGradle()).baseProjectClassLoaderLocked((InternalAction<Gradle>) project -> {
```

Comment:
ℹ️ If `Gradle.rootProject` callback uses BuildOperation infrastructure, as we have in integTest
https://github.com/gradle/gradle/blob/bbafd10381b97edc8f1961cf97f6cf04de733ce3/testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/BuildOperationNotificationsFixture.groovy#L57, then this Bridge has to be stopped a bit later, as soon as execution point of the `Gradle.rootProject` was shifted as well.

I'm not 100 confident in this change, since this is an internal API and not supposed to be used, but that `BuildOperationNotificationListenerRegistrar` is used by scan plugin.